feat:Load metadata & data from staff mapping

mutesasira opened the pull request:
This PR is to load metadata and data from staff mapping

- Updated useGetSelectedKeys to work with section-specific DataStore configs
- Added finalResult interface to TypeScript types with status field
- Made InfoPage content dynamic based on sectionType (student/staff)
- Made table title context-aware (Final Results vs Staff Re-enrollment)
- All program stages, status fields, and data elements now loaded from DataStore
- No hardcoded IDs - fully metadata-driven for both student and staff contexts

mutesasira committed:
feat:Load metadata & data from staff mapping

EdsonNhancale commented on src/hooks/config/useGetSelectedKeys.ts:
```diff
@@ -3,9 +3,13 @@ import { useDataStoreKey, useProgramsKeys } from "dhis2-semis-components";
+    const dataStoreArray = useDataStoreKey({ sectionType: sectionName });
     const programsValues = useProgramsKeys();
 
+    const dataStoreData = Array.isArray(dataStoreArray)
```
I think dataStoreArray always will be a object based on sectionName

mutesasira replied:
The current approach I used is the safest because it handles both cases, if useDataStoreKey returns an array (needs filtering) or a single object (already filtered by the external hook). 
The Array.isArray() check is some good defensive programming approach

EdsonNhancale replied:
I got you, have you found the cases that useDataStoreKey({ sectionType: sectionName }); returns an array?

mutesasira replied:
No, I have not found evidence that useDataStoreKey({ sectionType: sectionName }) returns an array. 
Since It's from the external package or library(if you call it that) dhis2-semis-components
I am passing sectionType as a parameter to it.
The hook is likely already filtering and returning the specific section's config object.

mutesasira replied:
Probably useDataStoreKey already returns a single object, not an array, making the Array.isArray() check unnecessary but harmless, i personally see it as a good to have

EdsonNhancale replied:
Yes, that's way I said you don't need to do that!

mutesasira replied:
unnecessary but harmless

mutesasira committed:
refactor(useGetSelectedKeys): remove unnecessary Array.isArray check

mutesasira committed:
feat(finalResult): add staff re-enrollment support with dynamic metadata loading

- Updated useGetSelectedKeys to work with section-specific DataStore configs
- Added finalResult interface to TypeScript types with status field
- Made InfoPage content dynamic based on sectionType (student/staff)
- Made table title context-aware (Final Results vs Staff Re-enrollment)
- All program stages, status fields, and data elements now loaded from DataStore
- No hardcoded IDs - fully metadata-driven for both student and staff contexts
